### PR TITLE
added the zk-proof writeup from concordium

### PR DESCRIPTION
**Pull request by wisbech**
Adding the excellent write up of approach to ZK-proofs that makes the registry operator less dependent on storing `m` and `r` values for pedersen-commitments/proofs. This method will significantly reduce the operational complexity of a registry and encapsulate domain knowledge.

Please feel free to change this document if the prerequisites have changed.

### CODE REVIEW

File: doc/concept/zk-proof.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+- We recommend using [Dalek Bulletproofs: A pure-Rust implementation of Bulletproofs using Ristretto](https://github.com/dalek-cryptography/bulletproofs) for these. Aggregation means that all ranges are proven within one proof.
+- Use the `Ed25519` curve and `G` a subgroup of prime order.
+
+#### Sum Proof
+Consider commitment `C_sum = C_1*...*C_n*(C)^-1` which is a commitment to zero if `m_1+...+m_n - m = 0`.
+
+In this case we have `C_sum = Com(0;r’) = h^{r’}` where `r’=r_1+...+r_n-r`.
+  
+Public Input:
+- The commitment `C_sum`
```
Comment (tschudid) [inline doc/concept/zk-proof.md:70]: The public input should also contain the commitments C_i for the summands. 
It is important for the FS-Sigma Protocol that the summand are input to the hash oracle.

**Review by tschudid — CHANGES_REQUESTED** [review]
Overall looks good to me. 
For the sum protocol it is important that the summand are part of the public input (which is used for the FS transform) as otherwise the proof becomes malleable.

File: doc/concept/zk-proof.md (hunk 1 above)
Comment (wisbech) [reply doc/concept/zk-proof.md:70]: @tschudid I am not quite sure how to make the correction myself to have the correct C_i notation - could you make a proposal? Just pick the line in the browser by clicking '+' and 'ctrl-g' or 'command-g' to make a change proposal as you see fit. 

Cheers

**Comment by wisbech** [comment]
Closing this PR and removing branch documentation/concepts/zk-proofs due to unfortunate rebasing

**Pull request closed without merging**